src/api/app/models/cloud/azure/configuration.rb

```diff
@@ -0,0 +1,44 @@
+module Cloud
+  module Azure
+    class Configuration < ApplicationRecord
+      validate :presence_of_fields
+      before_save :encrypt_credentials
+
+      belongs_to :user
+
+      def self.table_name_prefix
+        'cloud_azure_'
+      end
+
+      private
+
+      def encrypt_credentials
+        self.application_id = Crypto::Encrypt.cloud_upload_data(application_id) if application_id_changed?
+        self.application_key = Crypto::Encrypt.cloud_upload_data(application_key) if application_key_changed?
```

Comment:
Why not run the callback only `on: :update`?

Reply:
Is it necessary to encrypt application id as well? Is it considered as a secret?

Reply:
@ChrisBr I consider it as a secret. In case of a data breach, why exposing one of the credentials which is used for authentication? We promise to encrypt it so we should not do half things.

Reply:
I did not ask if you consider it as a secret but if MS does consider it as a secret :smile: What is written about it in the docs?